src/plugin/maven/src/main/kotlin/BaseMojo.kt

```diff
@@ -30,7 +41,33 @@ abstract class BaseMojo : AbstractMojo() {
         override fun error(string: String) = log.error(string)
     }
 
-    protected fun getFilesContent(): FilesContent = (File(input).listFiles() ?: arrayOf<File>())
+    protected fun getFilesContent() = input.split(",")
+        .flatMap {
+            when {
+                it.startsWith("classpath:") -> readFromClasspath(it.substringAfter("classpath:"))
+                else -> readFromFile(it)
+            }
+        }
+
+    private fun readFromClasspath(input: String): List<Pair<String, String>> = input
+        .let {
+            val file = File(it)
+            val classLoader = javaClass.classLoader
+            val inputStream = classLoader.getResourceAsStream(input)
+                ?: error("Could not find file: $it on the classpath.")
+            val content = inputStream.bufferedReader(Charsets.UTF_8).use { it.readText() }
+            val name = file.name.split(".").first()
+            return listOf(name to content)
+        }
```

Comment:
This suggests that you can't read from a directory in the classpath, only from one file directly. For symmetry with reading from file path I'd prefer to also allow reading from a classpath directory

Reply:
Mmm, would be nice how could you detect it is a file or directory in the classpath ?